Asterix/wfsc/wf_sensing_functions.py

```diff
@@ -196,17 +297,50 @@ def simulate_pw_difference(input_wavefront,
-
-        Ikplus = testbed.todetector_intensity(entrance_EF=input_wavefront,
-                                              voltage_vector=voltage_vector + Voltage_probe,
-                                              wavelengths=wavelength,
-                                              **kwargs)
+        # If we are in a polychromatic mode but we need monochromatic instensity
+        # we have to be careful with the normalization, because
+        # todetector_intensity is normalizing to polychromatic PSF by default
+        # At some point, we will need to do estimate at different
+        # WL than the one of the testbed (to do the AJ's trick where you scattered the WL
+        # of correction the most efficiently in the BW). We will have to be careful with normalization here
+        # and recode this part
+
+        if isinstance(wavelengths, (float, int)) and wavelengths in testbed.wav_vec:
+            # hard case : we are monochromatic for the probes, but polychromatic for the rest of images
+            # polychromatic = 'centralwl' or polychromatic = 'multiwl'
+            Ikmoins = testbed.todetector_intensity(
+                entrance_EF=input_wavefront,
+                voltage_vector=voltage_vector - Voltage_probe,
+                wavelengths=wavelengths,
+                in_contrast=False,
+                **kwargs) / testbed.norm_monochrom[testbed.wav_vec.tolist().index(wavelengths)]
+
+            Ikplus = testbed.todetector_intensity(
+                entrance_EF=input_wavefront,
+                voltage_vector=voltage_vector + Voltage_probe,
+                wavelengths=wavelengths,
+                in_contrast=False,
+                **kwargs) / testbed.norm_monochrom[testbed.wav_vec.tolist().index(wavelengths)]
+
+        elif np.all(wavelengths == testbed.wav_vec):
+            # easy case: we are monochromatic or polychromatic both for images and probes
+            # It's either a monochromatic correction or a polychromatic correction with
+            # polychromatic = 'broadband_pwprobes'
+            Ikmoins = testbed.todetector_intensity(entrance_EF=input_wavefront,
+                                                   voltage_vector=voltage_vector - Voltage_probe,
+                                                   wavelengths=wavelengths,
+                                                   **kwargs)
+
+            Ikplus = testbed.todetector_intensity(entrance_EF=input_wavefront,
+                                                  voltage_vector=voltage_vector + Voltage_probe,
+                                                  wavelengths=wavelengths,
+                                                  **kwargs)
+
+        else:
+            raise Exception(
+                """You are trying to do a pw_difference with wavelength parameters I don't understand. 
+                                Code it yourself in simulate_pw_difference and be careful with the normalization"""
+            )
```

Comment:
You don't need the triple quotes here.

Reply:
yeah but it allows me to put it on several line to be more readable

Reply:
not working for me, I get pylinted ! or maybe i don't understand what you mean
<img width="739" alt="Screenshot 2022-10-06 at 02 15 07" src="https://user-images.githubusercontent.com/22528508/194186230-29ee791e-ef49-44f7-9c00-4c5f8d660240.png">

Reply:
ok found a way ! 
<img width="698" alt="Screenshot 2022-10-06 at 02 27 31" src="https://user-images.githubusercontent.com/22528508/194187347-172a0d1e-d00a-4af8-9098-f46040209de5.png">